### PR TITLE
add support for favicon.ico

### LINKED ISSUE #9086: [FEATURE] add favicon support to Bokeh server
in bokeh/demo.bokeh.org#14 I reported that https://demo.bokeh.org doesn't have a [favicon](https://en.wikipedia.org/wiki/Favicon).  @bryevdv pointed out that nothing coming from a Bokeh server will have it set.

favicon support in bokeh server would help applications appear more "polished" to users because favicon is displayed by most browser clients in the 'tab' along with the <title> (from inside \<head\>).

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #9086
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

### CODE REVIEW

File: bokeh/server/tornado.py [hunk 1]
```diff
@@ -117,6 +118,8 @@ class BokehTornado(TornadoApplication):
         prefix (str, optional) :
             A URL prefix to use for all Bokeh server paths. (default: None)
 
+        icon_paath (str, optional) :
```
Comment (mattpap) [inline bokeh/server/tornado.py:121]:
```suggestion
        icon_path (str, optional) :
```

**Comment by bryevdv** [comment]
I've added more tests, so marking this as `ready`. cc @mattpap @tcmetzger 

I had considered adding support for `favicon.ico` in directory apps being picked up, but that will actually be a little complicated, and is at most a convenience on top of this. I'll save it for later if at all.

**Review by tcmetzger — COMMENTED** [review]
Thank you, I only have some very small text-related comments!

One general consideration: in docstrings and object/variable names, we are now mixing the terms "icons" and "favicons". I might make it easier to understand what we are talking about if we go with the term "favicon" consistently (differentiating the specific concept of favicon from more general icons that can be used in various other places).

**Comment by bryevdv** [comment]
Per call w/ @tcmetzger will rename to `--ico-path` to emphasize that option relates only to favicon***.ico*** files.

File: bokeh/server/tornado.py [hunk 2]
```diff
@@ -117,6 +118,8 @@ class BokehTornado(TornadoApplication):
         prefix (str, optional) :
             A URL prefix to use for all Bokeh server paths. (default: None)
 
+        icon_path (str, optional) :
+
```
[the remark this answers is not in the document]
Comment (bryevdv) [reply bokeh/server/tornado.py:123]: 2257bb2

**Comment by bryevdv** [comment]
@EfremBraun is this an acceptable start?

**Comment by EfremBraun** [comment]
Gorgeous! Yep, that's all I need to be happy. I see my site's icon showing up in the tab as it should.

The only thing I might request as a "nice to have but not necessary" feature would be for there to be a location in the directory tree that the `ico` file could be placed such that it's automatically found as the default setting, with the `bokeh.io` file being a backup in case that file is not present. That way I wouldn't have to set the `--ico-path` argument each time. But perhaps it's considered better practice or more typical Tornado server usage to do it the way that it's currently being done.


Just FYI: when I don't set the `--ico-path` argument, I get the error:
```
2021-11-14 08:14:39,094 Cannot start Bokeh server [ENOENT]: FileNotFoundError(2, 'No such file or │directory') 
```
This is because the `bokeh.ico` and `bokeh-dev.ico` files are not being copied to the install directory when I do `python setup.py install`. Not sure why; perhaps the build system needs data files to be stated explicitly in order for them to get copied over.

**Comment by bryevdv** [comment]
Manifest should be correct now. I had already thought of the directory automatic detection idea unfortunately it's just more complication than I can fit in at present.

**Comment by EfremBraun** [comment]
Changing the manifest fixed the issue in my development environment at least.

Thanks so much for taking care of this issue! Makes my application's website look much better.

**Pull request merged**